app/uk/gov/hmrc/tariffclassificationfrontend/controllers/PdfDownloadController.scala

```diff
@@ -49,10 +49,13 @@ class PdfDownloadController @Inject()(authenticatedAction: AuthenticatedAction,
 
   def applicationPdf(reference: String): Action[AnyContent] = authenticatedAction.async { implicit request =>
     caseService.getOne(reference) flatMap {
-      case Some(c: Case) =>
-        fileStore.getAttachments(c).flatMap { files =>
-          generatePdf(application_template(c, files), s"BTIApplication_$reference.pdf")
-        }
+      case Some(c: Case) => {
+        fileStore.getAttachments(c).flatMap(files =>
+          fileStore.getLetterOfAuthority(c).flatMap(letter =>
```

Comment:
Should this not be done in parallel (like the trader Ui)?

Reply:
I've changed into this approach which allows me easier to redirect NOT FOUND. Ultimately the for comprension is translating into the same structure. Is just nicer to read.

Reply:
You are right.  I thought the for comprehension structure automatically resulted in parallel execution.  My bad.     Would it not be better to re-structure like this https://alvinalexander.com/scala/how-use-multiple-scala-futures-in-for-comprehension-loop . (in both UIs).

Not critical I guess as generating the pdf takes far longer than getting the data together to make the request to the pdf-service and printing pdfs is not something that will be done very often.

Reply:
I'd use for comprehension here rather than `flatMap`

Reply:
I use for comprehension but I've change it in order to align both implementations just because **one of them use case Not found** and was easier in this way.

Reply:
`Easier != better` though @pllaurado

Reply:
Easier (not always means) better would be more accurate.